Open Judge System/OJS.Common/Extensions/FileHelpers.cs

```diff
@@ -56,7 +58,10 @@ public static string FindFirstFileMatchingPattern(string workingDirectory, strin
                     nameof(pattern));
             }
 
-            return ProcessModulePath(files[0]);
+            string discoveredFile = files
+                .OrderByDescending(f => new FileInfo(f).Length).First();
```

Comment:
Validation already exists and it throws an ArgumentException if no file matching the provided pattern is found.